src/qibo/tests/test_core_circuit_noise.py

```diff
@@ -160,3 +160,31 @@ def test_density_matrix_circuit_errors():
     c = Circuit(5)
     with pytest.raises(ValueError):
         c.add(gates.KrausChannel([((0,), np.eye(2))]))
+
+
+def test_circuit_add_sampling():
+    """Check measurements when simulating added circuits with noise"""
+    # Create random noisy circuit and add noiseless inverted circuit
+    gates_set = [gates.X, gates.Y, gates.Z, gates.H, gates.S, gates.SDG, gates.I]
+    circ = Circuit(1)
+    circ_no_noise = Circuit(1)
+    for _ in range(100):
+        new_gate = np.random.choice(gates_set)(0)
+        circ.add(gates.PauliNoiseChannel(0, pz=0.01))
+        circ.add(new_gate)
+        circ_no_noise.add(new_gate)
+        circ.add(gates.PauliNoiseChannel(0, pz=0.01))
```

Comment:
It doesn't really matter for testing purposes, but the last noise channel is outside the for loop in #563. Also it is fine to use a smaller circuit for testing (eg. less than 100 gates).

Reply:
Yes, I agree. I can lower it down to 10.